src/cacheRefresher.js

```diff
@@ -0,0 +1,185 @@
+        }
+
+        startAutoRefresh() {
+            this._intervalId = setInterval(() => this.callAndRefresh(), 10000); // every 10 seconds
+        }
+
+        callAndRefresh() {
+            const that = this;
+            const soapCall = this._client._prepareSoapCall("xtk:session", "GetModifiedEntities", true, this._connectionParameters._options.extraHttpHeaders);
+
+            if (this._lastTime === undefined) {
+                let storedTime = this._metadataCache.get("time");
+                if (storedTime != undefined) {
+                    this._lastTime = storedTime;
+                }
+            }
+            if (this._buildNumber === undefined) {
+                let storedBuildNumber = this._metadataCache.get("buildNumber");
+                if (storedBuildNumber != undefined) {
+                    this._buildNumber = storedBuildNumber;
+                }
+            }
+
+            // Use Json because xtk:schema does not work directly in DomUtil.parse(`<cache buildNumber="9469" lastModified="2022-06-30T00:00:00.000"><xtk:schema></xtk:schema></cache>`);
+            // due to the semi-colon character
+            var jsonCache;
+            if (this._lastTime === undefined || this._buildNumber === undefined) {
+                jsonCache = {
+                    [this._cacheSchema]: {}
+                }
+            } else {
+                jsonCache = {
+                    buildNumber: this._buildNumber,
+                    lastModified: this._lastTime,
+                    [this._cacheSchema]: {}
+                }
+            }
+
+            var xmlDoc = DomUtil.fromJSON("cache", jsonCache, 'SimpleJson');
+            soapCall.writeDocument("script", xmlDoc);
+
+            // Do a soap call GetModifiedEntities instead of xtksession.GetModifiedEnties because we don't want to go through methodCache 
+            // which can be wrong just after a build updgarde from a old version of acc that has not the method GetModifiedEntities and 
+            // a new version of acc that has the method GetModifiedEntities
+            return this._client._makeSoapCall(soapCall)
+                .then(() => {
+                    var doc = soapCall.getNextDocument();
+                    soapCall.checkNoMoreArgs();
+                    doc = that._client._toRepresentation(doc, 'xml');
+                    that._lastTime = DomUtil.getAttributeAsString(doc, "time"); // save time to be able to send it as an attribute in the next soap call
+                    that._buildNumber = DomUtil.getAttributeAsString(doc, "buildNumber");
+                    that.refresh(doc, that._cacheSchema);
+                    that._metadataCache.put("time", that._lastTime);
+                    that._metadataCache.put("buildNumber", that._buildNumber);
+                    Promise.resolve();
+                })
+                .catch((ex) => {
+                    if (soapCall.methodName == "GetModifiedEntities" && ex.errorCode == "SOP-330006") {
```

Comment:
It would be helpful to add a comment to explain why we do this, i.e. that the GetModifiedEntities method is not present in all builds and that if it's not, we'll disable the auto-refresh mechanism